### PR TITLE
Remove support for the unstable dir flag on relations.

### LINKED ISSUE #13919: Stabilize support for MSC3715
[MSC3715](https://github.com/matrix-org/matrix-doc/pull/3715) adds a `dir` parameter to `/relations`. This has now passed FCP and we should support the stable identifier.

* [x] Add support for the stable identifier.
* [ ] Remove support for the unstable identifier.

This was initially added to Synapse in #11926.

### PR DESCRIPTION
**Pull request by clokep**
Clients aren't using this, there's no way for clients to know it is even supported, and now that Matrix 1.4 is released we don't need it.

Fixes #13919

### CODE REVIEW

**Comment by DMRobertson** [comment]
> synapse/handlers/relations.py:120: error: Argument "limit" to "get_relations_for_event" of "RelationsWorkerStore" has incompatible type "Optional[int]"; expected "int"  [arg-type]

File: synapse/rest/client/relations.py [hunk 1]
```diff
@@ -60,17 +59,7 @@ async def on_GET(
         #
         # TODO Use PaginationConfig.from_request when the unstable parameter is
         #      no longer needed.
-        direction = parse_string(request, "dir", allowed_values=["f", "b"])
-        if direction is None:
-            if self._msc3715_enabled:
-                direction = parse_string(
-                    request,
-                    "org.matrix.msc3715.dir",
```
Comment (DMRobertson) [inline synapse/rest/client/relations.py:68]: To check: this is an unstable query parameter that's been stabilised, and we're now requiring the use of the stable version?

File: synapse/handlers/relations.py [hunk 2]
```diff
@@ -123,10 +117,10 @@ async def get_relations(
             room_id=room_id,
             relation_type=relation_type,
             event_type=event_type,
-            limit=limit,
-            direction=direction,
-            from_token=from_token,
-            to_token=to_token,
+            limit=pagin_config.limit,
```
Comment (DMRobertson) [inline synapse/handlers/relations.py:120]: mypy unhappy here. Is this just an `assert expr is not None`? (Having said that, I'm wondering if PaginationConfig.limit being None makes sense?)

File: synapse/streams/config.py [hunk 3]
```diff
@@ -42,10 +42,12 @@ async def from_request(
         cls,
         store: "DataStore",
         request: SynapseRequest,
-        raise_invalid_params: bool = True,
```
Comment (DMRobertson) [inline synapse/streams/config.py:45]: Was this just not used?

**Review by DMRobertson — COMMENTED** [review]
Seems sane! If we can get mypy happy it LGTM.

A few sanity checks:

File: synapse/streams/config.py (hunk 3 above)
Comment (clokep) [reply synapse/streams/config.py:45]: Yes, it was not used.

File: synapse/rest/client/relations.py (hunk 1 above)
Comment (clokep) [reply synapse/rest/client/relations.py:68]: Correct.

File: synapse/handlers/relations.py (hunk 2 above)
Comment (clokep) [reply synapse/handlers/relations.py:120]: Although it looks invasive enough that I might go with the assert for now?

Comment (DMRobertson) [reply synapse/handlers/relations.py:120]: Naively I'd expect "don't use a limit" to be represented by a lack of PaginationConfig.

Happy with an assert for now!

**Review by DMRobertson — APPROVED** [review]

**Pull request merged**